### PR TITLE
fix: Use topological (recursive) emptiness

### PR DESCRIPTION
**Pull request by ianthetechie**
I stumbled upon some other edge cases when checking compatibility against PostGIS. Basically the PG stance (also verified in JTS, so I assume GEOS is also the same) is that emptiness checks should be recursive based on the principle of "does this thing actually represent at least a point or does the structure simply collapse to nothing."

This showed up as I was working on an implementation for `ST_Dump` (PR soon :D) and trying to keep it native. After a bunch of test cases, I realized that PostGIS actually drops these, and there went the afternoon researching what we should do 😂

### CODE REVIEW

File: rust/geodatafusion/src/udf/native/accessors/is_empty.rs [hunk 1]
```diff
@@ -187,6 +205,27 @@ mod test {
                 false,
                 "non-empty geometry collection",
             ),
+            // Recursive emptiness (PostGIS / JTS topological semantics).
+            // A collection whose every leaf is empty is itself reported as empty,
+            // even though it has structurally non-zero children.
+            //
+            // NB: GeoArrow explicitly denies recursive collections,
+            // so we only need to test one level.
```
Comment (kylebarron) [inline rust/geodatafusion/src/udf/native/accessors/is_empty.rs]: It sounds like this only applies to GeometryCollection? Can any other supported geometry types have an empty child?

Comment (ianthetechie) [reply rust/geodatafusion/src/udf/native/accessors/is_empty.rs]: Good question... so in OGC terms, no, it actually applies to everything. MultiPoints can have a mix of empty and non-empty points, etc... My quick reading of the GeoArrow spec also seems to allow this. However, when writing up some extra tests for those, I discovered that the WKT parser we depend on only accepts mixed empty/non-empty multi-linestrings (https://github.com/georust/wkt/issues/111 seems to acknowledge this, but the symptom I see right now isn't a crash but a parse error). And multipolygons mixing empty and non-empty seem to crash, but those should also technically be allowed.

Comment (paleolimbot) [reply rust/geodatafusion/src/udf/native/accessors/is_empty.rs]: You could have a few WKB fixtures for this (this is what we do in SedonaDB). geoarrow.pyarrow's WKT parser should be able to generate these for you.

https://github.com/apache/sedona-db/blob/298911255f91e0d55146a8baf8e9b49aad404b35/rust/sedona-testing/src/fixtures.rs#L22-L29

Comment (ianthetechie) [reply rust/geodatafusion/src/udf/native/accessors/is_empty.rs]: Good call! I generated them with PostGIS `ST_AsBinary` and interpolated the literal, but same idea.

The multipolygon case I came up with is still broken due to an assumption in geoarrow-array. I'm not sure if there is a geoarrow rule forbidding that structure, but it works in PostGIS 🤷

**Review by kylebarron — APPROVED** [review]
Thanks!
